src/Microsoft.AspNetCore.Server.Kestrel.Core/Features/IHttpMinRequestBodyDataRateFeature.cs

```diff
@@ -1,18 +1,25 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+
 namespace Microsoft.AspNetCore.Server.Kestrel.Core.Features
 {
     /// <summary>
     /// Represents a minimum data rate for the request body of an HTTP request.
     /// </summary>
-    public interface IHttpRequestBodyMinimumDataRateFeature
+    public interface IHttpMinRequestBodyDataRateFeature
     {
         /// <summary>
         /// The minimum data rate in bytes/second at which the request body should be received.
-        /// Setting this property to null indicates no minimum data rate should be enforced.
+        /// Setting this property to zero effectively means no minimum data rate should be enforced.
         /// This limit has no effect on upgraded connections which are always unlimited.
         /// </summary>
-        MinimumDataRate MinimumDataRate { get; set; }
+        double BytesPerSecond { get; set; }
+
+        /// <summary>
+        /// The amount of time to delay enforcement of the minimum data rate.
```

Comment:
Will add ", starting at the time data is first read or written." Written as well since this same type will (likely) be used for the response rate.